input/templates/partials/catalog/reserve/initialize-map-scripts.hbs

```diff
@@ -0,0 +1,79 @@
+<script>
+  $(window).load(function() {
+    var addressList = $('.store-location-wrapper .addresses .list');
+    var addressItem = addressList.find('.item');
+    var coordsContainer = ".gm-computed-distance"
+    var coordValues = "data-value";
+    var activeClass = "address--active";
+
+    // change the location onClick
+    addressList.on('click', '.item', function() {
+      var changeLocationLatLng = $(this).find(coordsContainer).attr(coordValues);
+      var changeLocationParts = changeLocationLatLng.split(',');
+      var changeLocationLat = parseFloat(changeLocationParts[0]);
+      var changeLocationLng = parseFloat(changeLocationParts[1]);
+      setLocation(changeLocationLat, changeLocationLng);
+      addressItem.removeClass(activeClass);
+      $(this).addClass(activeClass);
+    });
+
+    // checks if google maps API is loaded
+    if (typeof google === 'object' && typeof google.maps === 'object') {
```

Comment:
What happens if on window.load google maps is not defined? Then geolocation will just not work? You probably need to ensure that it always available at this point or have some fallback like try reloading it?